src/main/java/dev/slimevr/vr/processor/skeleton/LegTweakBuffer.java

```diff
@@ -654,4 +700,133 @@ private float getRightFootLockLiklyHood() {
+		// if they did not we assume there is another force acting on the com
+		// and fall back to a low pressure prediction
+		if (detectOutsideForces(leftFootForce, rightFootForce))
+			return FORCE_VECTOR_FALLBACK;
+
+		// set the pressure to the force on each foot times the force to
+		// pressure scalar
+		leftFootPressure = leftFootForce.mult(FORCE_VECTOR_TO_PRESSURE).length();
+		rightFootPressure = rightFootForce.mult(FORCE_VECTOR_TO_PRESSURE).length();
+
+		// distance from the ground is a factor in the pressure
+		// using the inverse of the distance to the ground scale the
+		// pressure
+		float leftDistance = (leftFootPosition.y > leftFloorLevel)
+			? (leftFootPosition.y - leftFloorLevel)
+			: LegTweaks.NEARLY_ZERO;
+		leftFootPressure *= 1.0f / (leftDistance);
+		float rightDistance = (rightFootPosition.y > rightFloorLevel)
+			? (rightFootPosition.y - rightFloorLevel)
+			: LegTweaks.NEARLY_ZERO;
+		rightFootPressure *= 1.0f / (rightDistance);
+
+		// normalize the pressure values
+		float pressureSum = leftFootPressure + rightFootPressure;
+		leftFootPressure /= pressureSum;
+		rightFootPressure /= pressureSum;
+
+		return new float[] { leftFootPressure, rightFootPressure };
+	}
+
+	// preform a gradient descent to find the force vectors that best match the
+	// acceleration of the com
+	private void findForceVectors(Vector3f leftFootForce, Vector3f rightFootForce) {
+		int iterations = 100;
+		float stepSize = 0.01f;
+		// setup the temporary variables
+		Vector3f tempLeftFootForce1 = leftFootForce.clone();
+		Vector3f tempLeftFootForce2 = leftFootForce.clone();
+		Vector3f tempRightFootForce1 = rightFootForce.clone();
+		Vector3f tempRightFootForce2 = rightFootForce.clone();
+		Vector3f error;
+		Vector3f error1;
+		Vector3f error2;
+		Vector3f error3;
+		Vector3f error4;
+
+		for (int i = 0; i < iterations; i++) {
+			tempLeftFootForce1.set(leftFootForce);
+			tempLeftFootForce2.set(leftFootForce);
+			tempRightFootForce1.set(rightFootForce);
+			tempRightFootForce2.set(rightFootForce);
+
+			// get the error at the current position
+			error = centerOfMassAcceleration
+				.subtract(leftFootForce.add(rightFootForce).add(gravity));
+
+			// add and subtract the error to the force vectors
+			tempLeftFootForce1 = tempLeftFootForce1.mult(1.0f + stepSize);
+			tempLeftFootForce2 = tempLeftFootForce2.mult(1.0f - stepSize);
+			tempRightFootForce1 = tempRightFootForce1.mult(1.0f + stepSize);
+			tempRightFootForce2 = tempRightFootForce2.mult(1.0f - stepSize);
+
+			// get the error at the new position
+			error1 = centerOfMassAcceleration
+				.subtract(tempLeftFootForce1.add(rightFootForce).add(gravity));
+			error2 = centerOfMassAcceleration
+				.subtract(tempLeftFootForce2.add(rightFootForce).add(gravity));
+			error3 = centerOfMassAcceleration
+				.subtract(leftFootForce.add(tempRightFootForce1).add(gravity));
+			error4 = centerOfMassAcceleration
+				.subtract(leftFootForce.add(tempRightFootForce2).add(gravity));
```

Comment:
maybe these calculations should be in a separate function, my copy paste detector throws warnings at me here